util/clientmetric/clientmetric.go

```diff
@@ -7,18 +7,24 @@
 package clientmetric
 
 import (
+	"bytes"
+	"encoding/binary"
 	"fmt"
 	"io"
 	"sort"
+	"strings"
 	"sync"
 	"sync/atomic"
+	"time"
 )
 
 var (
-	mu          sync.Mutex
+	mu          sync.Mutex // guards vars in this block
 	metrics     = map[string]*Metric{}
-	sortedDirty bool
-	sorted      []*Metric
+	numWireID   int       // how many wireIDs have been allocated
+	lastDelta   time.Time // time of last call to EncodeLogTailMetricsDelta
```

Comment:
optional: would it make sense to attach the information for generating deltas to the particular log stream rather than being global? That would make it easier to send logs to two destinations concurrently if we want to do that in future.

Reply:
It might make sense, but I don't want to overdesign and complicate the code [yet] for something we might never use.